toolchain/check/type_structure.cpp

```diff
@@ -387,15 +402,24 @@ class TypeStructureBuilder {
                type_id.has_value()) {
       Push(type_id);
     } else {
-      Push(NonTypeValue());
+      Push(NonTypeValue{.type_id = context_->insts().Get(inst_id).type_id()});
     }
   }
 
   // Push the next step into the work queue.
   auto Push(WorkItem item) -> void { work_list_.push_back(item); }
 
   // Append a structural element to the TypeStructure being built.
-  auto AppendStructural(TypeStructure::Structural structural) -> void {
+  auto AppendStructural(TypeStructure::Structural structural,
```

Comment:
FWIW, this kind of feels like three different functions:

- `Concrete` and `ConcreteOpenParen`, which take a type and pushes it onto `concrete_types_`
- `Symbolic`, which pushes on `symbolic_type_indices_`
- `ConcreteCloseParen`, which only pushes on `structure_`